lib/Backends/NNPI/NNPIOptions.cpp

```diff
@@ -0,0 +1,437 @@
+       << "=" << useIceT_ << "\n";
+  desc << "  "
+       << options[InferOnDeviceOption][NNPIParamOptionEnvironmentVarName] << "="
+       << inferOnDevice_ << "\n";
+  desc << "  " << options[CompiledFileOption][NNPIParamOptionEnvironmentVarName]
+       << "=" << compiledFile_ << "\n";
+  desc << "  " << options[UseSymlowpOption][NNPIParamOptionEnvironmentVarName]
+       << "=" << useSymlowp_ << "\n";
+  desc << "  "
+       << options[DeviceVersionOption][NNPIParamOptionEnvironmentVarName] << "="
+       << deviceVersion_ << "\n";
+  desc << "  " << options[IceCoresOption][NNPIParamOptionEnvironmentVarName]
+       << "=" << iceCores_ << "\n";
+  desc << "  "
+       << options[CompilationDebugConfigFileOption]
+                 [NNPIParamOptionEnvironmentVarName]
+       << "=" << debugCompileConfigFile_ << "\n";
+  desc << "  "
+       << options[CompilationLogLevelOption][NNPIParamOptionEnvironmentVarName]
+       << "=" << compilationLogLevel_ << "\n";
+  desc << "  " << options[ShowVarsOption][NNPIParamOptionEnvironmentVarName]
+       << "=" << showVars_ << "\n";
+  return desc.str();
+}
+
+llvm::StringMap<std::string> NNPIDeviceOptions::getSupportedOptions() {
+  static llvm::StringMap<std::string> supportedOptions =
+      buildFilteredSupportedDescription({UseIceTOption, InferOnDeviceOption,
+                                         CompiledFileOption, NumOfWorkersOption,
+                                         DeviceIDOption, DeviceTraceOption,
+                                         ShowVarsOption},
+                                        NNPIOptions::getOptions());
+  return supportedOptions;
+}
+
+NNPIDeviceOptions::NNPIDeviceOptions(
+    const llvm::StringMap<std::string> *parameters) {
+  useIceT_ = NNPIOptions::getBoolVal<llvm::StringMap<std::string>>(
+      UseIceTOption, parameters);
+  inferOnDevice_ = NNPIOptions::getBoolVal<llvm::StringMap<std::string>>(
+      InferOnDeviceOption, parameters);
+  internalTesting_ = NNPIOptions::getStringVal<llvm::StringMap<std::string>>(
+                         InternalTestingOption, parameters) != "";
+  showVars_ = NNPIOptions::getBoolVal<llvm::StringMap<std::string>>(
+      ShowVarsOption, parameters);
+  compiledFile_ = NNPIOptions::getStringVal<llvm::StringMap<std::string>>(
+      CompiledFileOption, parameters);
+  deviceID_ = NNPIOptions::getIntVal<llvm::StringMap<std::string>>(
+      DeviceIDOption, parameters);
+  numWorkers_ = NNPIOptions::getIntVal<llvm::StringMap<std::string>>(
+      NumOfWorkersOption, parameters);
+  enabledDeviceTraceing_ =
+      NNPIOptions::getBoolVal<llvm::StringMap<std::string>>(DeviceTraceOption,
+                                                            parameters);
+// Todo: remove if 1 section once nnpiDeviceGetInfo changes are implemented.
+#if 1
+  deviceMemory_ = 16000000;
+#else
+  deviceMemory_ = NNPIOptions::getUnsignedVal<llvm::StringMap<std::string>>(
```

Comment:
Why can't this just be enabled now, and override `deviceMemory_ ` if specified? I.e. just check if the specified value is the default 0 and if not then use it.